src/snowflake/snowpark/stored_procedure.py

```diff
@@ -792,7 +792,8 @@ def _do_register_sp(
             if package_name not in self._session._packages:
                 packages = list(self._session._packages.values()) + [this_package]
         else:
-            if not any(package_name in p for p in packages):
+            package_names = [p if isinstance(p, str) else p.__name__ for p in packages]
+            if not any(p.startswith(package_name) for p in package_names):
```

Comment:
Will be removed from this PR once https://github.com/snowflakedb/snowpark-python/pull/1401 is merged